Jellyfin.Drawing.Skia/SkiaEncoder.cs

```diff
@@ -111,6 +111,113 @@ public static SKEncodedImageFormat GetImageFormat(ImageFormat selectedFormat)
+            using var canvas = surface.Canvas;
+            using var paint = new SKPaint
+            {
+                FilterQuality = SKFilterQuality.High,
+                IsAntialias = isAntialias,
+                IsDither = isDither
+            };
+
+            var kernel = new float[9]
+            {
+                0,    -.1f,    0,
+                -.1f, 1.4f, -.1f,
+                0,    -.1f,    0,
+            };
+
+            var kernelSize = new SKSizeI(3, 3);
+            var kernelOffset = new SKPointI(1, 1);
+
+            paint.ImageFilter = SKImageFilter.CreateMatrixConvolution(
+                kernelSize,
+                kernel,
+                1f,
+                0f,
+                kernelOffset,
+                SKShaderTileMode.Clamp,
+                true);
+
+            canvas.DrawBitmap(
+                source,
+                SKRect.Create(0, 0, source.Width, source.Height),
+                SKRect.Create(0, 0, targetInfo.Width, targetInfo.Height),
+                paint);
+
+            return surface.Snapshot();
+        }
+
+        /// <summary>
+        /// Decode an image.
+        /// </summary>
+        /// <param name="path">The filepath of the image to decode.</param>
+        /// <param name="forceCleanBitmap">Whether to force clean the bitmap.</param>
+        /// <param name="orientation">The orientation of the image.</param>
+        /// <param name="origin">The detected origin of the image.</param>
+        /// <returns>The resulting bitmap of the image.</returns>
+        internal SKBitmap? Decode(string path, bool forceCleanBitmap, ImageOrientation? orientation, out SKEncodedOrigin origin)
+        {
+            if (!File.Exists(path))
+            {
+                throw new FileNotFoundException("File not found", path);
+            }
+
+            var requiresTransparencyHack = _transparentImageTypes.Contains(Path.GetExtension(path));
+
+            if (requiresTransparencyHack || forceCleanBitmap)
+            {
+                using var codec = SKCodec.Create(NormalizePath(path));
+                if (codec == null)
+                {
+                    origin = GetSKEncodedOrigin(orientation);
+                    return null;
+                }
+
+                // create the bitmap
+                var bitmap = new SKBitmap(codec.Info.Width, codec.Info.Height, !requiresTransparencyHack);
+
+                // decode
+                _ = codec.GetPixels(bitmap.Info, bitmap.GetPixels());
+
+                origin = codec.EncodedOrigin;
+
+                return bitmap;
+            }
+
+            var resultBitmap = SKBitmap.Decode(NormalizePath(path));
+
+            if (resultBitmap == null)
+            {
+                return Decode(path, true, orientation, out origin);
+            }
+
+            // If we have to resize these they often end up distorted
+            if (resultBitmap.ColorType == SKColorType.Gray8)
+            {
+                using (resultBitmap)
```

Comment:
Wouldn't resultBitmap.Dispose() be more efficient in this instance?

Reply:
I haven't changed anything here, all of this LoC is just relocating the code block to fix the mentioned warning (also fixed in ), which consisted in moving the internal stuff above the private.

I don't have the knowledge to answer this question properly but, afaik, using ``using`` is the same as a ``try/finally`` block where ``Dispose()`` is called at the end.

Reply:
Did see it was in the original code.

Agree, but as there doesn't seem to be any reason/need for a _try/finally_ block around the _return_, a _dispose_ here would seem to be more efficient.